Yoast/wordpress-seo
# P1-1 Output correct schema based on Schema settings

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Page type and, if applicable, Article type set on the indexable or search appearance settings are now taken into account when generating Schema graphs on the front-end.
* Article type setting is hidden for custom post types that do not support an author, because the current implementation of Yoast's schema requires an Author graph piece to be present for Article schema to be valid.
* "None" is no longer an option for the Page type setting, because it has the potential to break schema, and as an option had very limited usefulness.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Add a custom post type and have author support enabled (easiest way is using Custom Post Type UI plugin), and create a new post for it. Verify that in the metabox schema tab, you CAN set article type.
* Add a custom post type and have author support disabled (easiest way is using Custom Post Type UI plugin), and create a new post for it. Verify that in the metabox schema tab, you CANNOT set article type.
* In search appearance, verify that the CPT with author support CAN set article, and the CPT without author support CANNOT.
* Play around with setting Page types and Article types in the search appearance settings, and leaving the metabox settings on default. Verify that your selected types appear in the schema output on the frontend (save post, view post, view source).
* Now verify that the metabox settings override the search appearance settings.

## UI changes

## Documentation

## Quality assurance


Fixes P1-1
